subprojects/common/result/src/main/kotlin/com/avito/android/Result.kt

```diff
@@ -27,30 +23,36 @@ sealed class Result<T> {
         is Failure -> func()
     }
 
-    inline fun <R> map(func: (value: T) -> R): Result<R> = when (this) {
+    inline fun <R> map(
+        mapFailure: (Throwable) -> Throwable = { failure -> failure },
+        mapSuccess: (value: T) -> R
```

Comment:
let's switch positions for the sake of consistency:

```
1. success
2. failure
```

In other functions, we stick to this order.

Reply:
I can't switch because this order helps to write code like
`result.map { }` because we have lambda as a last parameter